### PR TITLE
Fix _dataframe sparseness

### PR DESCRIPTION
**Pull request by danielballan**
The muxer's internal `_dataframe` as not being built properly. Data keys belonging to the same Event should go into a single row, not separate rows. This fixes that and adds several tests.

### CODE REVIEW

**Comment by danielballan** [comment]
New commit adds tests and `logger.debug` messages.

File: dataportal/muxer/data_muxer.py [hunk 1]
```diff
@@ -233,12 +233,13 @@ def append_event(self, event):
...
-        for name, data_dict in event.data.items():
-            # Both scalar and nonscalar data will get stored in the DataFrame.
-            # This may be optimized later, but it might not actually help much.
-            self._data.append({name: event.data[name][0]})
-            self._timestamps.append({name: event.data[name][1]})
-            self._time.append(event.time)
+        # Both scalar and nonscalar data will get stored in the DataFrame.
+        # This may be optimized later, but it might not actually help much.
+        self._data.append(
+            {name: data[0] for name, data in event.data.items()})
```
Comment (tacaswell) [inline dataportal/muxer/data_muxer.py]: This is someplace where `six.iteritems` makes sense

Comment (danielballan) [inline dataportal/muxer/data_muxer.py]: I don't think that's right. This works in 2 and 3 the same.

Comment (tacaswell) [inline dataportal/muxer/data_muxer.py]: In 2 this is going to make a list of the items, in 3 this is going to return an iterator.  `six.iteritems(event.data)` returns an iterator is both cases.

Comment (danielballan) [inline dataportal/muxer/data_muxer.py]: It works the same, but I agree it is slightly less efficient in 2 than 3.

**Pull request merged**